### PR TITLE
feat: dispatch vtk scroll event

### PR DESCRIPTION
**Pull request by dannyrb**
Per our conversation yesterday, I need a way to know when and which a VTK Viewport is scrolled. It appears the default event is blocked/prevented by VTK internally. This attempts to re-emit the event as a `vtkscrollevent` so I can consume it somewhere else.

The addition of a `uid` on the "api" object, and on the `interactorStyle` are to provide a way for me to determine, for a given `viewportIndex`, if it was the event's emitter.

Related: https://github.com/OHIF/Viewers/pull/1139

### CODE REVIEW

File: src/VTKViewport/View2D.js [hunk 1]
```diff
@@ -248,6 +253,19 @@ export default class View2D extends Component {
 
       this.props.onCreated(api);
     }
+
+    // Set tracking id on interactorStyle model
+    // If we set it immediately, this value is blown out by model's default
+    setTimeout(() => {
+      this.setUid(uid);
+    }, 1000);
```
Comment (dannyrb) [inline src/VTKViewport/View2D.js]: I tried setting the `uid`'s value when using the interactor style's `newInstance` factory function, but the value is temporarily set and then blown out.

So I instead tried setting it using a public setter here, but experience a similar result.

This is my 🤷‍♂ attempt, but it "works".

Comment (JamesAPetts) [reply src/VTKViewport/View2D.js]: Can't you just pass it down via configuration when we set the interactorStyle via `api.setInteractorStyle` in OHIF?

i.e.
// Pseudocode
```
apis.forEach(api => {
 const uid = api.uid;
api.setInteractorStyle({iStyle, configuration: { uid }}); // Assuming iStyle is defined above + add any other relavent config
});
```

Comment (dannyrb) [reply src/VTKViewport/View2D.js]: I'll give that a try and report back ^_^

Comment (dannyrb) [reply src/VTKViewport/View2D.js]: But we set the interactorStyle for MPR slice internally for View2D? OHIF never sets it.

Comment (dannyrb) [reply src/VTKViewport/View2D.js]: Ah. Inheritance. Fun fun. Ready for you to re-review.

**Comment by dannyrb** [comment]
@JamesAPetts, best way to document the addition of a new event?

**Comment by dannyrb** [comment]
@JamesAPetts I've assigned you to review this thing's sister:

https://github.com/OHIF/Viewers/pull/1139

^ Won't be valid until this is merged.

**Comment by JamesAPetts** [comment]
I've merged this for now, we can review event propogation again if/when a more mature api exists.

**Pull request merged**